tss-esapi-sys/regenerate-bindings.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 Contributors to the Parsec project.
+# SPDX-License-Identifier: Apache-2.0
+
+# Cross compile the `tss-esapi` crate (and its dependencies) for Armv7 and Aarch64
+# In order to cross-compile the TSS library we need to also cross-compile OpenSSL
+
+set -euf -o pipefail
+
+OPENSSL_VERSION="OpenSSL_1_1_1j"
+
+cross-compile-openssl() {
+    # Prepare directory for cross-compiled OpenSSL files
+    mkdir -p /tmp/openssl-$1
+    export INSTALL_DIR=/tmp/openssl-$1
+
+    pushd /tmp/openssl
+    # Compile and copy files over
+    ./Configure $2 shared --prefix=$INSTALL_DIR --openssldir=$INSTALL_DIR/openssl --cross-compile-prefix=$1-
+    make clean
+    make depend
+    make -j$(nproc)
+    make install
+    popd
+
+    export INSTALL_DIR=
+}
+
+cross-compile-tpm2-tss() {
+    # Prepare directory for cross-compiled TSS lib
+    # `DESTDIR` is used in `make install` below to set the root of the installation paths.
+    # The `./configure` script accepts a `--prefix` input variable which sets the same root,
+    # but also adds it to the paths in `.pc` files used by `pkg-config`. This prevents the 
+    # use of `PKG_CONFIG_SYSROOT_DIR`.
+    mkdir -p /tmp/tpm2-tss-$1
+    export DESTDIR=/tmp/tpm2-tss-$1
+    # Set sysroot to be used by the `pkg-config` wrapper
+    export SYSROOT=/tmp/tpm2-tss-$1
+
+    pushd /tpm2-tss
+    # Compile and copy files over
+    ./configure --build=x86_64-pc-linux-gnu --host=$1 --target=$1 CC=$1-gcc \
+        LIBCRYPTO_CFLAGS="-I/tmp/openssl-$1/include" LIBCRYPTO_LIBS="-L/tmp/openssl-$1/lib -lcrypto"
+    make clean
+    make -j$(nproc)
+    make install
+    popd
+
+    export DESTDIR=
+}
+
+# Download cross-compilers
+apt update
+apt install -y gcc-multilib
+apt install -y gcc-arm-linux-gnueabi
+apt install -y gcc-aarch64-linux-gnu
+
+# Download OpenSSL source code
+if [ ! -d "/tmp/openssl" ]; then
+    pushd /tmp
+    git clone https://github.com/openssl/openssl.git --branch $OPENSSL_VERSION
+    popd
+fi
+
+# Regenerate bindings for x86_64-unknown-linux-gnu
+cargo clean
+cargo build --features generate-bindings
```

Comment:
It would be nice to make the `generate-bindings` feature copy the bindings to their right location and use that directly, to not have to copy it manually. That would be good for all of our `-sys` crate actually.

Reply:
ah yeah true... And in some places those directories are read-only